src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/client/CompleteAwaitingDcnProcessingTest.java

```diff
@@ -0,0 +1,165 @@
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.ccd.CcdAuthenticator;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.ccd.CcdAuthenticatorFactory;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.ccd.CcdClient;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class CompleteAwaitingDcnProcessingTest {
+
+    private static final String AWAITING_DCN_PROCESSING_FIELD_NAME = "awaitingPaymentDCNProcessing";
+    private static final String COMPLETE_AWAITING_PROCESSING_EVENT_ID = "completeAwaitingPaymentDCNProcessing";
+    private static final String EVENT_SUMMARY = "Complete payment DCN processing";
+
+    @Mock
+    private CoreCaseDataApi ccdApi;
+
+    @Mock
+    private CcdAuthenticatorFactory ccdAuthenticatorFactory;
+
+    private CcdClient ccdClient;
+
+    @BeforeEach
+    void setUp() {
+        ccdClient = new CcdClient(ccdApi, ccdAuthenticatorFactory);
+    }
+
+    @Test
+    void should_start_and_submit_event_in_ccd() {
+        // given
+        String exceptionRecordId = "1231244243242343";
+        String service = "bulkscan";
+        String jurisdiction = "BULKSCAN";
+        String serviceToken = "serviceToken1";
+        String userId = "userId1";
+        String userToken = "userToken1";
+        String eventToken = "eventToken1";
+
+        setUpCcdAuthenticationFactory(userId, userToken, serviceToken);
+
+        StartEventResponse startEventResponse = mockStartEventResponse(eventToken);
+        given(ccdApi.startEventForCaseWorker(any(), any(), any(), any(), any(), any(), any()))
+            .willReturn(startEventResponse);
+
+        // when
+        ccdClient.completeAwaitingDcnProcessing(exceptionRecordId, service, jurisdiction);
+
+        // then
+        verify(ccdAuthenticatorFactory).createForJurisdiction(jurisdiction);
+
+        String expectedCaseTypeId = "BULKSCAN_ExceptionRecord";
+
+        verify(ccdApi).startEventForCaseWorker(
+            userToken,
+            serviceToken,
+            userId,
+            jurisdiction,
+            expectedCaseTypeId,
+            exceptionRecordId,
+            COMPLETE_AWAITING_PROCESSING_EVENT_ID
+        );
```

Comment:
you _could_ just put those in mocking stage

Reply:
I think it's most explicit this way.